### PR TITLE
Update support for search zone automatically

### PR DESCRIPTION
**Pull request by dongxifu**

### CODE REVIEW

File: context/context.go [hunk 1]
```diff
@@ -71,7 +71,20 @@ func SetupContext(c *Config) error {
 	qsService, err := service.Init(curQingStorConfig)
 	check.ErrorForExit("Init QingStor config error", err)
 
-	Bucket, err = qsService.Bucket(c.BucketName, c.Zone)
+	Zone := ""
+
+	if c.Zone == "" {
+		output, _ := qsService.ListBuckets(nil)
```
Comment (Xuanwo) [inline context/context.go]: 这样做不好，用户不一定使用的是他自己的 bucket。

**Comment by pengsrc** [comment]
“Update support” in commit message should be changed to “Support”.

File: context/context.go [hunk 2]
```diff
@@ -71,7 +71,23 @@ func SetupContext(c *Config) error {
 	qsService, err := service.Init(curQingStorConfig)
 	check.ErrorForExit("Init QingStor config error", err)
 
-	Bucket, err = qsService.Bucket(c.BucketName, c.Zone)
+	Zone := ""
+
+	if c.Zone == "" {
+		Zones := [2]string{"pek3a", "sh1a"}
```
Comment (Xuanwo) [inline context/context.go]: We have private cloud without other zone names.

File: context/context.go [hunk 3]
```diff
@@ -71,7 +103,15 @@ func SetupContext(c *Config) error {
 	qsService, err := service.Init(curQingStorConfig)
 	check.ErrorForExit("Init QingStor config error", err)
 
-	Bucket, err = qsService.Bucket(c.BucketName, c.Zone)
+	Zone := ""
+
+	if c.Zone == "" {
+		Zone = GetZone(c)
```
Comment (pengsrc) [inline context/context.go]:
``` Go
if c.Zone == "" {
    c.Zone = GetZone(c)
}
```

File: context/context.go [hunk 4]
```diff
@@ -17,6 +17,9 @@
 package context
 
 import (
+	"fmt"
+	"net/http"
+	"strings"
```
Comment (pengsrc) [inline context/context.go]: Missing new line here.

File: context/context.go [hunk 5]
```diff
@@ -33,6 +36,25 @@ var (
...
+					    c.QingStor.Port)
+
+	response, err := http.Head(url)
+	check.ErrorForExit("Request error", err)
+	if(err != nil) {
+		return "", err
+	}
+
+	//URL for example: https://bucket.zone.example.com
+	NewURL := response.Request.URL.String()
```
Comment (pengsrc) [inline context/context.go]: Should start with lowercase.

File: context/context.go [hunk 6]
```diff
@@ -71,7 +93,16 @@ func SetupContext(c *Config) error {
 	qsService, err := service.Init(curQingStorConfig)
 	check.ErrorForExit("Init QingStor config error", err)
 
-	Bucket, err = qsService.Bucket(c.BucketName, c.Zone)
+	zone := ""
+
+	if c.Zone == "" {
```
Comment (pengsrc) [inline context/context.go]:
``` Go
zone := c.Zone
if zone == "" {
    zone, err = GetZone(c)
    check.ErrorForExit("Get zone error", err)
}

**Pull request merged**